src/driver/op.rs

```diff
@@ -23,9 +24,23 @@ pub(crate) struct Op<T: 'static> {
 
 pub(crate) trait Completable {
     type Output;
+    fn update(self, result: io::Result<u32>, flags: u32) -> Update<Self::Output, Self>
+    where
+        Self: Sized,
+    {
+        Update::Finished(self.complete(result, flags))
+    }
     fn complete(self, result: io::Result<u32>, flags: u32) -> Self::Output;
 }
 
+pub(crate) enum Update<F, M> {
+    /// The operation has finished
+    Finished(F),
+
+    /// The operation expects more completion events
+    More(M),
```

Comment:
Could this M be renamed Op? Still trying to get my head around why there isn't a variant added to the Lifecycle to handle the new variant of More, but maybe I'm just stuck thinking about things the way I worked them in a different fork for the streaming operations.

Reply:
These are just maximally generic, and match Finished and More (F/M).